src/models/syncRule.ts

```diff
@@ -79,16 +84,25 @@ export default class SyncRule {
         filters: SyncRuleTableFilters,
         pagination: TablePagination
     ): Promise<TableList> {
-        const { targetInstanceFilter = null } = filters || {};
+        const { targetInstanceFilter = null, enabledFilter = null, lastExecutedFilter = null } =
+            filters || {};
         const data = await getPaginatedData(d2, dataStoreKey, filters, pagination);
-        return targetInstanceFilter
-            ? {
-                  ...data,
-                  objects: _.filter(data.objects, e =>
-                      e.builder.targetInstances.includes(targetInstanceFilter)
-                  ),
-              }
-            : data;
+        return {
+            ...data,
+            objects: _(data.objects)
+                .filter(rule =>
+                    targetInstanceFilter
+                        ? rule.builder.targetInstances.includes(targetInstanceFilter)
+                        : true
+                )
+                .filter(rule => (enabledFilter ? rule.enabled === enabledFilter : true))
+                .filter(rule =>
+                    lastExecutedFilter
+                        ? moment(lastExecutedFilter).isSameOrBefore(rule.lastExecuted)
```

Comment:
Really minor, but for rules without lastExecuted, this conditions returns true (`now.isSameOrBefore(undefined)` returns true, who knows why). Adding `rule.lastExecuted && ...` fixes it.

Reply:
See https://github.com/EyeSeeTea/metadata-synchronization/pull/134/commits/8623619bcdad92a6b23fb04d68617e54497dd811